examples/meta/src/evaluation/cross_validation_pipeline.sg

```diff
@@ -0,0 +1,32 @@
+File f_feats_train = csv_file("../../data/classifier_binary_2d_linear_features_train.dat")
+File f_feats_test = csv_file("../../data/classifier_binary_2d_linear_features_test.dat")
+File f_labels_train = csv_file("../../data/classifier_binary_2d_linear_labels_train.dat")
+File f_labels_test = csv_file("../../data/classifier_binary_2d_linear_labels_test.dat")
+
+#![create_features]
+Features feats_train = features(f_feats_train)
+Features feats_test = features(f_feats_test)
+Labels labels_train = labels(f_labels_train)
+Labels labels_test = labels(f_labels_test)
+#![create_features]
+
+#![create_pipeline
+Transformer subMean = transformer("PruneVarSubMean")
+Machine svm = machine("LibLinear")
+
+PipelineBuilder builder = pipeline()
+builder.over(subMean)
+Pipeline pipeline = builder.then(svm)
```

Comment:
Why does the API contain these two builder and pipeline concepts? What about just a pipeline where steps are added (and of course the order of addition matters).

Reply:
the idea is to separate construction of pipeline into a single class so that the pipeline object is immutable, and then we don't need to verify elements of pipeline everytime

Reply:
This is pretty much the same thing that got me confused about the builder vs the pipeline.

Reply:
the second one, `Pipeline` can be `Machine` here because that's enough, we use it as machine in xval. however, if we want to get elements in the pipeline, we still need `Pipeline` type for now

Reply:
what about adding a factory for it? So users can call
`PipelineMachine(machine)`? 
Alternatively, for the c++ folks, there is `as`

Reply:
I understand that there might be mutability reason justifying two types. Why both part of the interface though?